### PR TITLE
Tasks-Dragg And Drop

### PR DESCRIPTION
**Pull request by mohamedsaeid1998**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,8 @@
     "moment": "^2.29.4",
     "react": "^18.2.0",
     "react-bootstrap": "^2.9.1",
+    "react-dnd": "^16.0.1",
```
Comment (Ahmed-Abou-Emran) [inline package.json:21]: if one of these packages are not used anymore, you can remove it by running the command:
`npm uninstall "packageName"`

File: src/Components/Done/Done.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }))
+
+  const isActive = canDrop && isOver;
+  let backgroundColor = "rgba(49, 89, 81, 0.90)";
+  if (isActive) {
+    backgroundColor = "darkgreen";
+  } else if (canDrop) {
+    backgroundColor = "darkkhaki";
+  }
+  console.log(isOver, canDrop);
```
Comment (Ahmed-Abou-Emran) [inline src/Components/Done/Done.tsx]: Don't forget to comment or remove the logs before merging into the main branch.

File: src/Components/Todo/Todo.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { useDrop } from 'react-dnd';
+import { Task } from '..';
+import './Todo.module.scss';
+import { IEmpTasks } from '@/Interfaces/EmployeeTasks';
+
+
+
+
+
+const Todo = ({ AllTasks,getTasksData }: IEmpTasks) => {
```
Comment (Ahmed-Abou-Emran) [inline src/Components/Todo/Todo.tsx:14]: We have three components: ToDo, InProgress, and Done. They're quite similar, and updating them requires changes in three separate files. To simplify, we can create a single column component, serving as the main reference. This not only streamlines edits but also makes adapting to potential new statuses more straightforward, avoiding code complexity.

File: src/Pages/Tasks/Tasks.tsx [hunk 4]
```diff
@@ -46,10 +54,46 @@ const Tasks = () => {
...
+  })
+  // console.log(AllTasks);
+
+  const getTasksData = () => {
+    return baseUrl.get(`/api/v1/Task?pageSize=1000&pageNumber=1`, {
+      headers,
+    })
+      .then((res) => {
+        console.log(res.data.data);
+        setAllTasks({
```
Comment (Ahmed-Abou-Emran) [inline src/Pages/Tasks/Tasks.tsx:77]: you can improve the readability of this part by doing the following:

```javascript
setAllTasks({
  todo: filterTasksByStatus(res, "ToDo"),
  inProgress: filterTasksByStatus(res, "InProgress"),
  done: filterTasksByStatus(res, "Done")
});
const filterTasksByStatus = (data, status) => data?.data?.filter((task) => task.status === status);
```

**Review by Ahmed-Abou-Emran — CHANGES_REQUESTED** [review]
### Required Changes

- [x] commenting or removing the logs before merging into the main branch.
- [x] Improving readability of tasks filtration

### Optional

- [ ]  Abstracting the `Todo`, `InProgress`, `Done` components into a Column component for better code reusability and  maintainability.

**Comment by mohamedsaeid1998** [comment]
The necessary changes have been completed, and I will make the optional task later in a separate branch

**Pull request merged**